src/main/java/seedu/address/logic/commands/EditInternshipCommand.java

```diff
@@ -0,0 +1,218 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_INTERNSHIPS;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.internship.CompanyName;
+import seedu.address.model.internship.Internship;
+import seedu.address.model.internship.InternshipId;
+import seedu.address.model.internship.InternshipRole;
+import seedu.address.model.internship.InternshipStatus;
+import seedu.address.model.internship.InterviewDate;
+import seedu.address.model.person.PersonId;
+
+/**
+ * Edits the details of an existing internship in the address book.
+ */
+public class EditInternshipCommand extends Command {
+
+    public static final String COMMAND_WORD = "editi";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the internship identified "
+            + "by the index number used in the displayed internship list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com";
```

Comment:
Should have been company name, role, status, contact person index, interview date

Reply:
Oh yeah, thanks